### PR TITLE
Generate and bundle minimized BTFs

### PR DESCRIPTION
**Pull request by ISauve**
### What does this PR do?

Creates new CI jobs:
a) `build_processed_btfhub_archive`, which is responsible for fetching, processing and pushing to S3 a collection of BTF files for platforms/kernel versions which are known to not have BTF information by default.
b) `generate_minimized_btfs_x64`/`generate_minimized_btfs_arm`, which are responsible for downloading the stored BTF collections and generating collections of minimized BTFs which have been tailored to the given bpf CO-RE program(s). The minimized BTF collections are then bundled into the datadog-agent package.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: omnibus/config/software/system-probe.rb [hunk 1]
```diff
@@ -37,6 +39,7 @@
     copy "#{ENV['SYSTEM_PROBE_BIN']}/tcp-queue-length.c", "#{install_dir}/embedded/share/system-probe/ebpf/runtime/"
     copy "#{ENV['SYSTEM_PROBE_BIN']}/clang-bpf", "#{install_dir}/embedded/bin/clang-bpf"
     copy "#{ENV['SYSTEM_PROBE_BIN']}/llc-bpf", "#{install_dir}/embedded/bin/llc-bpf"
+    command "tar -xf #{ENV['SYSTEM_PROBE_BIN']}/minimized-btfs.tar.gz -C #{install_dir}/embedded/share/system-probe/ebpf/co-re/btf/"
```
Comment (brycekahle) [inline omnibus/config/software/system-probe.rb:42]: Akin to my comment https://github.com/DataDog/datadog-agent/pull/13962#discussion_r1007213261 I think it would be best to leave this as a tarball.

File: .gitlab/deps_build.yml [hunk 2]
```diff
@@ -127,3 +127,45 @@ build_vcpkg_deps:
...
+    # Clean up platform names to match the names we get at runtime from gopsutil
+    - mv btfs-amd64/amzn btfs-amd64/amazon
+    - mv btfs-arm64/amzn btfs-arm64/amazon
+    - mv btfs-amd64/oracle-linux btfs-amd64/oracle
+    - mv btfs-arm64/oracle-linux btfs-arm64/oracle
+    # Store results in S3
+    - tar -czf btfs-arm64.tar.gz btfs-arm64
+    - tar -czf btfs-amd64.tar.gz btfs-amd64
+    - $S3_CP_CMD btfs-arm64.tar.gz $S3_PERMANENT_ARTIFACTS_URI/btfs-arm64.tar.gz
+    - $S3_CP_CMD btfs-amd64.tar.gz $S3_PERMANENT_ARTIFACTS_URI/btfs-amd64.tar.gz
```
Comment (brycekahle) [inline .gitlab/deps_build.yml:166]: You can probably clean this up, by iterating over the architectures with:
```
for arch in amd64 arm64; do
...
done
```

File: tasks/system_probe.py [hunk 3]
```diff
@@ -999,3 +999,45 @@ def tempdir():
...
+):
+    """
+    Given an input directory containing compressed full-sized BTFs, generates an identically-structured
+    output directory containing compressed minimized versions of those BTFs, tailored to the given
+    bpf program(s).
+    """
+
+    ctx.run(f"mkdir -p {output_dir}")
+    for root, dirs, files in os.walk(source_dir):
+        path_from_root = root[len(source_dir) + 1 :]
```
Comment (brycekahle) [inline tasks/system_probe.py:1019]:
```suggestion
        path_from_root = os.path.relpath(root, source_dir)
```

File: tasks/system_probe.py [hunk 4]
```diff
@@ -999,3 +999,45 @@ def tempdir():
...
+
+        for dir in dirs:
+            output_subdir = os.path.join(output_dir, path_from_root, dir)
+            ctx.run(f"mkdir -p {output_subdir}")
+
+        for file in files:
+            if not file.endswith(".tar.xz"):
+                continue
+
+            btf_filename = file[: len(file) - 7]
```
Comment (brycekahle) [inline tasks/system_probe.py:1029]:
```suggestion
            btf_filename = file[:-len(".tar.xz")]
```

File: tasks/system_probe.py [hunk 5]
```diff
@@ -999,3 +999,45 @@ def tempdir():
         yield dirpath
     finally:
         shutil.rmtree(dirpath)
+
+
+@task
+def generate_minimized_btfs(
```
[the remark this answers is not in the document]
Comment (ISauve) [reply tasks/system_probe.py:1005]: About 4-8 minutes!

File: .gitlab/deps_build.yml (hunk 2 above)
Comment (ISauve) [reply .gitlab/deps_build.yml:166]: My gripe with this is that I wouldn't be able to put comments, so I think it would actually be a little hard to follow what's happening

Comment (brycekahle) [reply .gitlab/deps_build.yml:166]: Fair. This isn't crucial, and we can clean it up later with a script or something.

File: tasks/system_probe.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (brycekahle) [reply tasks/system_probe.py:1005]: Interesting. We may want to explore using parallelism to get that time down. This is in the critical path for other ebpf-related CI jobs.

[the remark this answers is not in the document]
Comment (ISauve) [reply tasks/system_probe.py:1005]: That's definitely something we can look into

**Review by brycekahle — APPROVED** [review]

**Review by iglendd — APPROVED** [review]

**Review by amenasria — APPROVED** [review]
LGTM !
